src/main/scala/com/lightbend/kafkalagexporter/Domain.scala

```diff
@@ -33,21 +33,21 @@ object Domain {
     def apply(tuples: (TopicPartition, LookupTable.Point)*): PartitionOffsets = Map(tuples: _*)
   }
 
-  class TopicPartitionTable private(limit: Int, var tables: Map[TopicPartition, LookupTable.Table]) {
-    def apply(tp: TopicPartition): LookupTable.Table = {
-      tables = tables.updated(tp, tables.getOrElse(tp, LookupTable.Table(limit)))
+  class TopicPartitionTable private(var tables: Map[TopicPartition, Either[LookupTable.MemoryTable, LookupTable.RedisTable]], config: ConsumerGroupCollector.CollectorConfig) {
```

Comment:
Can you create a base `Table` type for the lookup table implementations instead of using `Either`? It would cleanup call sites so it's not necessary to match on left and right.